src/lib/core/datetime.c

```diff
@@ -586,7 +626,7 @@ datetime_datetime_sub(struct interval *res, const struct datetime *lhs,
 	return interval_interval_sub(res, &inv_rhs);
 }
 
-bool
+int
 interval_interval_sub(struct interval *lhs, const struct interval *rhs)
```

Comment:
I think it's best to add changes of these functions to the commit where they were introduced. Looks like it would simplify a lot of things.